### PR TITLE
TS-5056 Implement nonrecoverable error mechanism

### PR DESCRIPTION
**Pull request by danobi**
Change `Emergency()` to terminate the current process with status
code UNRECOVERABLE_EXIT.

Also change traffic_manager to listen for the UNRECOVERABLE_EXIT
status code. If heard, then TM will not try to restart TS from
that point forward.

This was designed so that traffic_server could call Emergency(..)
in the event of a nonrecoverable error such as a bad config file.
No amount of TS rebooting will fix a bad config, so we might as well
have TM wait for human intervention.

Note that if traffic_cop or traffic_manager calls Emergency(),
nothing totally unexpected will happen since the only visible change
from this patch is the status code.

### CODE REVIEW

File: lib/ts/ink_error.cc [hunk 1]
```diff
@@ -61,19 +61,38 @@ ink_fatal(const char *message_format, ...)
...
+}
+
+void
+ink_emergency(const char *message_format, ...)
+{
+  va_list ap;
+
+  va_start(ap, message_format);
+  ink_emergency_va(message_format, ap);
+  va_end(ap);
```
Comment (SolidWallOfCode) [inline lib/ts/ink_error.cc]: Hmmm. Hasn't the process already exited by this point? Maybe best to be safe, though.

**Comment by zwoop** [comment]
This needs to run clang-format again. Please rebase and push the PR again, and I'll land it.

**Comment by shinrich** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
I did this  by hand, to squash the two commits.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1308/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1202/ for details.

**Pull request closed without merging**